File: webapp/src/app/types/enums/ContractStatus.ts
export enum ContractStatus {
	ACTIVE = "Aktive",
	INACTIVE = "Inaktive",
	TERMINATED = "Beendet",
	BLOCKED = "Gesperrt",
	IN_PROGRESS = "In Bearbeitung",
	ANY = "Alle",
}

export const ContractStatusList = Object.entries(ContractStatus)
.map(([key, value]) => ({key, value}))

File: server/src/main/java/com/school_system/enums/school/ContractStatus.java
package com.school_system.enums.school;

import com.fasterxml.jackson.annotation.JsonValue;

public enum ContractStatus {

    ACTIVE("ACTIVE"),
    INACTIVE("INACTIVE"),
    TERMINATED("TERMINATED"),
    BLOCKED("BLOCKED"),
    IN_PROGRESS("IN_PROGRESS"),
    UNKNOWN("UNKNOWN"),
    ANY("Alle");

    private final String value;
    ContractStatus(String value){

        this.value = value;
    }
    @JsonValue
    public String getValue() {
        return value;
    }
}
